mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -298,11 +298,7 @@ class LinearRegressionModel private[ml] (
    */
   // TODO: decide on a good name before exposing to public API
   private[regression] def evaluate(dataset: DataFrame): LinearRegressionSummary = {
-    val t = udf { features: Vector => predict(features) }
-    val predictionAndObservations = dataset
-      .select(col($(labelCol)), t(col($(featuresCol))).as($(predictionCol)))
-
-    new LinearRegressionSummary(predictionAndObservations, $(predictionCol), $(labelCol))
+    new LinearRegressionSummary(transform(dataset), $(predictionCol), $(labelCol))
```

Comment:
I had recommended switching to transform, but I just realized that a user could potentially disable the output columns we need.  In that case, I'm really not sure what we should do.  I guess I'd say:
* Easy option: Check to make sure predictionCol is defined and not the empty string, and throw an exception if not.
* Nicer option: If predictionCol is disabled (empty string), then generate a column name like "prediction_[randomNumber]", copy the model with this new predictionCol set, and use it for the summary.

Either is OK with me, but it'd be cool if you have time to do the nicer option.  : )